src/tools/test/test_status_codes.js

```diff
@@ -0,0 +1,73 @@
+const fs = require("fs-extra");
+const fetch = require("node-fetch");
+const convert = require('xml-js');
+
+const base_url = "http://127.0.0.1:8080";
+
+let failures = 0;
+let passes = 0;
+
+const test_status_code = async (page,status,location) => {
+
+  if (location == undefined) {
+    location = null;
+  } else {
+    location = base_url + location;
+  }
+
+  try {
+
+    // Don't follow redirects
+    const options = {
+      redirect: 'manual'
+    }
+
+    const response = await fetch(base_url + page, options);
+
+    if (response.status === status && response.headers.get('location') === location) {
+      console.log('Success - expected:', status, 'got:',response.status, 'for page:', page);
+      passes++;
+    } else {
+      console.error('Failed - expected:', status, 'got:',response.status, 'for page:', page);
+      failures++;
+    }
+
+  } catch (error) {
+    console.error('Error - expected:', status, 'for page:', page);
+    console.log(error);
+    failures++;
+  }
+};
+
+const test_sitemap_pages = async (page,code) => {
+  const xml = await fs.readFile(`templates/sitemap.xml`, 'utf-8');
+  const sitemap = JSON.parse(convert.xml2json(xml,{compact: true}));
+  const urls = sitemap['urlset']['url'];
+  for ( var url in urls ) {
+    page = urls[url]['loc']['_text'];
```

Comment:
Should we trim any whitespaces, if present? (XML text node might be in a different line, which may not be the case if we know how we are generating the sitemap, but I guess there is not harm to harden this):

```suggestion
    page = urls[url]['loc']['_text'].trim();
```

Reply:
Good call. We generate the sitemap, but no harm to add this in case we change the format.